pkg/assembler/nodes.go

```diff
@@ -16,9 +16,161 @@
 package assembler
 
 // ArtifactNode is a node that represents an artifact
-// TODO: implement GuacNode
-type ArtifactNode struct{}
+type ArtifactNode struct {
+	Name     string
+	Checksum string
+}
+
+func (an ArtifactNode) Type() string {
+	return "Artifact"
+}
+
+func (an ArtifactNode) Properties() map[string]interface{} {
+	properties := make(map[string]interface{})
+	properties["name"] = an.Name
+	properties["checksum"] = an.Checksum
+	return properties
+}
+
+func (an ArtifactNode) Attributes() []string {
+	return []string{"name", "checksum"}
+}
+
+func (an ArtifactNode) IdentifiableAttributes() [][]string {
+	// An artifact can be uniquely identified by checksum
+	return [][]string{{"checksum"}}
+}
 
 // AttestationNode is a node that represents an attestation
-// TODO: implement GuacNode
-type AttestationNode struct{}
+type AttestationNode struct {
+	// TODO(mihaimaruseac): Unsure what fields to store here
+	File     string
```

Comment:
Is this meant to be the contents of the file? A path? URI?

Reply:
Should be a path. Changed to `FilePath`.

In my demo for neo4j I used the path of the attestation as the only attribute for this node, but not completely sure that this is what we need.